### ISSUE TITLE
QuickOpen crash

### ISSUE THREAD

**Issue by bert88sta** [extr0py/oni#125]
I get this error:

Error: Error: ENOENT: no such file or directory, stat '/Users/Bret/.wine/dosdevices/d:'

running from a non-git (my home) directory, I'd assume there's some sort of issue with how oni handles not being able to open certain files.

**Comment by extr0py** [comment]
I'm able to reproduce too with my non-git home directory (just a different file shows up EBUSY: hiberfile.sys). 

Seems like the `recursive-readdir` library we're using at the moment will fail as soon as it hits an issue, and not return a list of files. Couple of options for fixing this:
1) Use an alternative library that handles these cases more robustly (`shelljs` might be a good choice - I'd like to bring this in for some other scenarios as well - it has an ls command that might work better)
2) Use a default command for UNIX / Windows that doesn't require an additional node library

Also, we might want to update the Menu component to handle asynchronicity better - right now, we synchronously populate the menu which blocks. It'd be nice if we showed a spinner or something and items as they populate. I'll log a separate task for this shortly.

**Comment by keforbes** [comment]
> Seems like the recursive-readdir library we're using at the moment will fail as soon as it hits an issue, and not return a list of files.

Someone created a [fix](https://github.com/jergason/recursive-readdir/issues/42#issuecomment-269523863) for this but the author of `recursive-readdir` [refuses](https://github.com/jergason/recursive-readdir/pull/43#issuecomment-264189392) to merge it in.  I tried following the directions for manually installing that person's `#patch-2` but my `npm` just hangs when I try installing it.

I tried getting `shelljs` to work but it wouldn't connect to neovim.  I didn't have any compilation errors and everything looked like it should've worked but I would just get an empty electron window when I tried launching oni.  Looks like I was hitting an error in `bundle.js` for:
```
require('./commands.json').forEach(function (command) {
```
saying "`__webpack_require__(...).forEach is not a function`" though I have no idea how replacing `recursive-readdir` with `shelljs` could lead to that error.

So yeah, I guess I just wanted to complain.  I'm still playing with this.
